src/handlers/graphical.rs

```diff
@@ -231,7 +232,21 @@ impl GraphicalReportHandler {
                 let opts = textwrap::Options::new(width)
                     .initial_indent(&initial_indent)
                     .subsequent_indent(&rest_indent);
-                writeln!(f, "{}", textwrap::fill(&error.to_string(), opts))?;
+                match error {
+                    ErrorKind::Diagnostic(diag) => {
+                        let mut inner = String::new();
+
+                        // Don't print footer for inner errors
+                        let mut inner_renderer = self.clone();
+                        inner_renderer.footer = None;
+                        inner_renderer.render_report(&mut inner, diag)?;
+
+                        writeln!(f, "{}", textwrap::fill(&inner, opts))?;
```

Comment:
The idea is that if a developer chooses to go into giving a complete error experience they do not have to go and implement their own printer and instead can re-use the existing infrastructure.

As a concrete example I have the following error tree possible:

- User gives a configuration file with a variable amount of components and how to initialize them
    - Each component then gets loaded and can fail for various reasons, so that users can easily fix their errors all errors are collected and exposed as `related`. 
        - Each of these errors are a Diagnostic so that users get complete information with `source_code`s etc

An example output of this:

![image](https://user-images.githubusercontent.com/1631166/167258051-75e674f0-6247-48c2-b796-f3e0bd687f09.png)

Now, this screenshot is without this patch, but once added I will be able to attach more info to the inner error and thus have a clear hierarchy of what happened and how the user could fix it.